問題を外部から回答可能にし、親コンポーネントに回答を保存するようにした。Front/15

x0y14 opened the pull request:
<!--
必ず画面右のメニューからReviewers,Labels,Projectsを設定してください
-->

## 変更点
- Question4taku、QuestionAnaume、QuestionCollectionに個別に保存していたユーザーの回答データを、Question.vueで保存するようにしました。
- 上に伴い回答データの形式を変更しました。

## 対応するissue

<!-- closed #issue番号 のように記載してください -->
closed #15

x0y14 committed:
回答ページを追加

x0y14 committed:
ページ内の表示を変更

x0y14 committed:
問題保存形式を変更

x0y14 committed:
gitignoreにauth.jsonを追加

x0y14 committed:
スコア表示画面に遷移するための関数を作成

x0y14 committed:
fix yarn lock

x0y14 committed:
コンフリクトを解消

nanagami1369 commented:
`gitignoreにauth.jsonを追加` について
私の別のプルリクに対応するために用意するのはわかるのですが、
自分のタスクと関係の無いコミットをするのは無しでお願いします。

auth.jsonはすでに別のコミットでdevelop branchの.gitignoreに追加済みです。

自分だけのgitignoreを作るなら`{$project-root}/.git/info/exclude`に書けばできますので🙏

nanagami1369 commented on yarn.lock:
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
空のyarn.lockが追加されています。

nanagami1369 commented:
` 問題保存形式を変更 `のcommitでpackage.jsonからfirebaseuiが消えるのも、package-look.jsonが消えるのも関連性が見えません~~予測不能です~~。
コミット内容と実際の作業内容に乖離が生じています。
ご注意をお願いいたします🙇‍♀️

nanagami1369 commented on front/src/components/Question.vue:
```diff
@@ -1,17 +1,21 @@
       <h3>unknown : {{ questionData.questionType }}</h3>
     </div>
+    <br />
+    <button v-on:click="getUserAnswersAndGoScorePage()">答え合わせ</button>
```
ここだけ v-onが使われていて他が@を使った省略記号が使われています。
@に統一をお願いします。

nanagami1369 commented on front/src/components/Question.vue:
```diff
@@ -28,7 +32,57 @@ export default {
+      }
+      console.log(JSON.stringify(this.answerData))
+    },
+    sayhello() {
+      alert('hello')
+    },
```
使われていない関数があります。

nanagami1369 commented on front/src/components/Question.vue:
```diff
@@ -28,7 +32,57 @@ export default {
+  methods: {
+    // todo
+    getUserAnswersAndGoScorePage() {
+      for (let i = 0; i < this.userAnswers.length; i++) {
+        // 問題
+        for (let j = 0; j < this.userAnswers[i].length; j++) {
+          // 回答
+          this.answerData[i][j].userAnswer = this.userAnswers[i][j]
+        }
+      }
```
`i`と`j`じゃ読めん
二次元配列の`i`と`j`の変数名の慣例ですが、`i`はindexですが、その次の`j`はアルファベットで`i`の次は`j`という数学的な変数の考え方でつけられたものです。

というわけで配列の中身が予想できる変数名でお願いします。
例えば〇〇indexとか座標ならx,yなどの変数名が考えられます。

nanagami1369 replied:
特に補完機能が聞きづらい環境なので変数名が命です。
よろしくおねがいします。

nanagami1369 commented on front/src/components/Question.vue:
```diff
@@ -28,7 +32,57 @@ export default {
+        this.userAnswers.push([])
+        this.answerData[i] = []
+        for (
+          let j = 0;
+          j < this.questionData.questions[i].answers.length;
+          j++
+        ) {
+          this.answerData[i].push({
+            questionAnswer: this.questionData.questions[i].answers[j],
+            userAnswer: null,
+          })
```
`i`と`j`じゃ読めん……

x0y14 committed:
レビューいただいた部分を修正しました。
- vueの省略形が使われていない部分を修正
- 一文字変数を識別可能なものに
- テスト用に追加されていた未使用関数の削除
- ルートから未使用のyarn.lockを削除

nanagami1369 reviewed (approved):
OK
